### PR TITLE
Build system

### PR DESCRIPTION
**Pull request by ptomato**
Adds a simple Meson build system and fixes a few compiler / GJS warnings.

https://phabricator.endlessm.com/T16203

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+include_path = join_paths(meson.current_source_dir(), 'webhelper')
+tests_environment = environment()
+tests_environment.prepend('GJS_PATH', include_path)
+tests_environment.prepend('GI_TYPELIB_PATH', meson.current_build_dir())
+tests_environment.prepend('LD_LIBRARY_PATH', meson.current_build_dir())
+tests_environment.set('WEBHELPER_UNINSTALLED_EXTENSIONS_DIR',
+    meson.current_build_dir())
+tests_environment.set('TOP_SRCDIR', meson.current_source_dir())
+
+args = [jasmine.path()[0], '--no-config', '--tap']
```
Comment (ebassi) [inline meson.build]: `jasmin.path()` is not an array (tested with Meson 0.39.1), so you cannot access the first element. Removing the `[0]` made it work.

Comment (ptomato) [inline meson.build]: On Meson 0.37.1, removing the `[0]` gives me this error:

> Meson encountered an error in file meson.build, line 97, column 4:
> Command line arguments must be strings

It's also contrary to what the Meson documentation says about the `path()` method: https://github.com/mesonbuild/meson/wiki/Reference-manual#external-program-object Maybe this is a regression in the newer version?

Comment (ebassi) [inline meson.build]: It's likely; I've opened https://github.com/mesonbuild/meson/issues/1547 to track it.

**Comment by ebassi** [comment]
Merged, now that the Meson issue has been settled, and that we have Meson 0.39.1 in our build.

**Pull request merged**